src/ui/Search.js

```diff
@@ -0,0 +1,148 @@
+import React, {Component} from 'react';
+import Modal from 'react-modal';
+import {UP, DOWN, ESC, ENTER, PGUP, PGDN, F3} from '../keycode';
+
+// TODO: Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
+// Modal.setAppElement('#yourAppElement');
+
+export default class extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {showSearchModal: false, searchString: '', searchMode: 0};
+    this.searchBoxNode = null;
+  }
+
+  handleKeyModal = e => {
+    if (e.keyCode === ENTER || e.keyCode === ESC) { // enter or escape
+      this.setState({showSearchModal: false});
+
+      // NOTE(Oak): getActiveNode uses document.activeElement, but right now
+      // our focus is on the modal, so we use setTimeout to wait for the modal to close
+      // first
+      setTimeout(() => {
+        if (this.state.searchString === '') return;
+        this.props.searchModes[this.state.searchMode].initSearch(this.state.searchString);
+      });
```

Comment:
How long is the timer?

Reply:
I intend it to be 0 (which is the same as not specifying). This is a trick to append it to the event queue.

Reply:
Minor nit: can you explicitly write in the 0?

Reply:
Done.